torch/_torch_docs.py

```diff
@@ -1785,6 +1785,40 @@ def merge_dicts(*dicts):
 Alias for :func:`torch.clamp`.
 """.format(**common_args))
 
+add_docstr(torch.column_stack,
+           r"""
+column_stack(tensors, *, out=None) -> Tensor
+
+Creates a new 2-dimensional tensor by horizontally stacking the tensors in :attr:`tensors`.
+
+:attr:`tensors` must be composed entirely of 1-dimensional and 2-dimensional tensors whose first dimension is the same size.
```

Comment:
I apologize, but I was mistaken about this previously. There's not a restriction that the tensors be zero, one, or two dimensions or that the output will always be 2D. Your updated implementation and the tests make this clear.

I'd like to propose this doc update (it needs formatting), please let me know what you think of it, @RockingJavaBean:

"Creates a new tensor by horizontally stacking the tensors in :attr:\`tensors\`. 

Equivalent to torch.hstack(tensors), except each zero or one dimensional tensor `t` in tensors is first reshaped into a (t.numel(), 1) column before being stacked horizontally."

I think this is accurate (and a slight clarification about the shape of the columns). I looked at the NumPy implementation of this function, too, to validate:

https://github.com/numpy/numpy/blob/912790315033d8a2b8d525eb36db410a851ca1f4/numpy/lib/shape_base.py#L613

https://github.com/numpy/numpy/blob/e4630a2f5273307983b6ddca9b72aa0cf5a65784/numpy/core/shape_base.py#L287

In both cases `concatenate(arrs, 1)` is called, since column_stack avoids the first case in hstack (which has a special case for 1 dim arrays).

Reply:
I sincerely appreciate the kind sharing of the insights and gists of the `column_stack` function.
I think the description your proposed is really accurate and it inspired me to further simplify the transform code.
```c++
auto transform_lambda = [](const Tensor& input) -> Tensor {
  // reshape 0D or 1D tensor t into (t.numel(), 1)
  if (input.dim() <= 1) {
    return input.reshape({input.numel(), 1});
  }
  return input;
};
```

I also formatted the proposed description, below is the screenshot of the `column_stack` documentation rendered by my local machine.
![column_stack_doc](https://user-images.githubusercontent.com/40605299/96888536-52c34900-14b8-11eb-9f30-f5b5ade883dc.png)